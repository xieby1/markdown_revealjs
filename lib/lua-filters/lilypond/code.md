The filter should leave code elements (like this: `putStrLn "hello, world!"`)
and code blocks (like this:

```lua
function foo()
  return "foo"
end
```

) alone if they aren't tagged with the `lilypond` class.

It should also ignore elements that have the `ly-norender` class: `relative
c'`{.lilypond .ly-norender}.
