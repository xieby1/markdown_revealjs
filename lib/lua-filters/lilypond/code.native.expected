[ Para
    [ Str "The"
    , Space
    , Str "filter"
    , Space
    , Str "should"
    , Space
    , Str "leave"
    , Space
    , Str "code"
    , Space
    , Str "elements"
    , Space
    , Str "(like"
    , Space
    , Str "this:"
    , Space
    , Code ( "" , [] , [] ) "putStrLn \"hello, world!\""
    , Str ")"
    , SoftBreak
    , Str "and"
    , Space
    , Str "code"
    , Space
    , Str "blocks"
    , Space
    , Str "(like"
    , Space
    , Str "this:"
    ]
, CodeBlock
    ( "" , [ "lua" ] , [] )
    "function foo()\n  return \"foo\"\nend"
, Para
    [ Str ")"
    , Space
    , Str "alone"
    , Space
    , Str "if"
    , Space
    , Str "they"
    , Space
    , Str "aren\8217t"
    , Space
    , Str "tagged"
    , Space
    , Str "with"
    , Space
    , Str "the"
    , Space
    , Code ( "" , [] , [] ) "lilypond"
    , Space
    , Str "class."
    ]
, Para
    [ Str "It"
    , Space
    , Str "should"
    , Space
    , Str "also"
    , Space
    , Str "ignore"
    , Space
    , Str "elements"
    , Space
    , Str "that"
    , Space
    , Str "have"
    , Space
    , Str "the"
    , Space
    , Code ( "" , [] , [] ) "ly-norender"
    , Space
    , Str "class:"
    , Space
    , Code
        ( "" , [ "lilypond" , "ly-norender" ] , [] ) "relative c'"
    , Str "."
    ]
]
